core/trino-main/src/main/java/io/trino/sql/planner/ConnectorExpressionTranslator.java

```diff
@@ -149,6 +219,63 @@ public SqlToConnectorExpressionTranslator(Map<NodeRef<Expression>, Type> types)
             return Optional.of(new Constant(null, typeOf(node)));
         }
 
+        @Override
+        protected Optional<ConnectorExpression> visitFunctionCall(FunctionCall node, Void context)
+        {
+            if (!isComplexExpressionPushdown(session)) {
+                return Optional.empty();
+            }
+
+            // Currently, we treat node as a builtin function.
+            // TODO Distinguish between builtin, plugin-provided and runtime-added functions.
+            if (node.getFilter().isPresent() || node.getOrderBy().isPresent() || node.getWindow().isPresent() || node.getNullTreatment().isPresent() || node.isDistinct()) {
+                return Optional.empty();
+            }
+
+            String functionName = ResolvedFunction.extractFunctionName(node.getName());
+
+            if (LiteralFunction.LITERAL_FUNCTION_NAME.equalsIgnoreCase(functionName)) {
+                ExpressionInterpreter interpreter = new ExpressionInterpreter(node, plannerContext, session, types);
+                Object value = interpreter.evaluate();
+                verify(!(value instanceof Expression), "Literal function did not evaluate to a constant: %s", node);
```

Comment:
You can now use `ExpressionUtils.constantExpressionEvaluatesSuccessfully()` for this.

Reply:
Since I need the actual value, I've created a new method - `ExpressionUtils.evaluateConstantExpression()`